src/libutil/hash.hh

```diff
@@ -167,6 +167,21 @@ HashResult hashPath(HashType ht, const Path & path,
  */
 Hash compressHash(const Hash & hash, unsigned int newSize);
 
+/**
+ * Parse a string representing a hash format.
+ */
+HashFormat parseHashFormat(std::string_view hashFormatName);
+
+/**
+ * std::optional version of parseHashFormat that doesn't throw error.
+ */
+std::optional<HashFormat> parseHashFormatOpt(std::string_view hashFormatName);
+
+/**
+ * The reverse of parseHashFormat.
+ */
+std::string_view printHashFormat(HashFormat hashFormat);
+
```

Comment:
These should have unit tests that they round-trip. Ideally also a rapidcheck property test too.

Reply:
Round-trip test added.

I'm not familiar with rapidcheck property check. Are there examples of such tests? I'm still trying to figure out what https://github.com/emil-e/rapidcheck/blob/master/doc/properties.md would like to explain.

Reply:
There are examples of such tests, look at the store path tests.